packages/firestore/src/local/persistence_promise.ts

```diff
@@ -259,4 +243,25 @@ export class PersistencePromise<T> {
     }
     return p;
   }
+
+  /**
+   * Given an iterable, call the given function on each element in the
+   * collection and wait for all of the resulting concurrent
+   * PersistencePromises to resolve.
+   */
```

Comment:
I would make it more explicit that it runs in parallel:

```
/**
   * Given an iterable, call the given function on each element in
   * parallel and wait for all of the resulting concurrent
   * PersistencePromises to resolve.
   */
```

Reply:
I think your update is actually subtly wrong. The callback is called on each element in sequence,  not in parallel. The resulting promises run in parallel.